hw/ip/kmac/data/kmac.hjson

```diff
@@ -315,23 +376,34 @@
       ]
     } // R: STATUS
     { name: "ENTROPY_PERIOD"
-      desc: '''Entropy Refresh Period.
-
-      This register determines the refresh period. The value is the number of
-      clock cycles. In every entropy period, the entropy generation logic in
-      KMAC requests new entropy to EDN and feed the value into LFSR seed.
-
-      If 0, the entropy generation logic does not refresh its seed. The
-      software manually updates the seed by writing into "ENTROPY_SEED"
-      registers.
+      desc: '''Entropy Timer Periods.
       '''
       swaccess: "rw"
       hwaccess: "hro"
       regwen: "CFG_REGWEN"
       fields: [
-        { bits: "31:0"
-          name: "period"
-          desc: "entropy period in clock cycles"
+        { bits: "15:0"
+          name: "entropy_timer"
+          desc: '''Entropy period in clock cycles This register determines
+          the refresh period. The value is the number of clock cycles. In
+          every entropy period, the entropy generation logic in KMAC requests
+          new entropy to EDN and feed the value into LFSR seed.
+
+          If 0, the entropy generation logic does not refresh its seed. The
+          software manually updates the seed by writing into "ENTROPY_SEED"
+          registers. '''
+        }
+        { bits: "31:16"
+          name: "wait_timer"
+          desc: '''EDN request wait timer.
+
+          The entropy module in KMAC waits up to this field in clock cycles
+          after it sends request to EDN module. If the timer expires, the
+          entropy module moves to an error state and notifies to the system.
+
+          If 0, the entropy module waits the EDN response always. If EDN does
+          not respond in this configuration, the software shall reset the IP.
```

Comment:
Does this part mean that only a software reset will get us out of this state?

Reply:
Yes. But still it may not resolve the issue, as the request already sent to EDN but ack not arrive. May need to reset EDN too.

Reply:
yeah sounds good. I guess the idea is that if edn locks up, we might have bigger issues.